### ISSUE TITLE
export --compression none generates TAR but the filename is tar.gz

### ISSUE THREAD

**Issue by nobuto-m** [lxc/lxd#11543]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 22.04
 * The output of "lxc info" or if that fails:
   * Kernel version: `5.15.0-67-generic #74-Ubuntu`
   * LXD version: 5.12-c63881f
   * Storage backend in use: zfs/ceph

# Issue description

This is a very cosmetic issue. `lxc export test-c-1 --compression none` generates a backup in TAR format but the actual filename is backup.tar.gz.

Of course, the file name can be specified explicitly as `lxc export test-c-1 --compression none backup.tar`

# Steps to reproduce

 1. `lxc launch ubuntu:jammy test-c-1`
 2. `lxc export test-c-1 --compression none`
 3. `file backup.tar.gz` -> `POSIX tar archive`

**Comment by stgraber** [comment]
Yeah, that one has been bugging me for a little while. Part of the problem is that we may not know what the algorithm is as the default can be configured server-side.

I think the right approach for when no filename is provided should be:
 - Temporarily use `<instance name>.backup`
 - Scan for the magic after download
 - Rename to `<instance name>.<extension`

**Comment by ChrisSmith2** [comment]
Hi!
I'm a computer science student at the University of Texas at Austin taking a Virtualization course and one of our assignments is to contribute to an open source project. Would I be able to work on resolving this issue?

**Comment by tomponline** [comment]
Thanks @ChrisSmith2 assigned to you.
